### ISSUE TITLE
Look into replacing Tomcat with Jetty

### ISSUE THREAD

**Issue by jpschewe** [jpschewe/fll-sw#296]
Jetty is used by Jenkins and other projects that need an embedded server. We should look at Jetty and see if it makes sense to switch. The primary concern is if this makes it easier for the user to get fll-sw running.

**Comment by jpschewe** [comment]
I really would like to be able to embed the web server in such a way that `JAVA_HOME ` doesn't need to be set as this is a bit of a pain as Java versions get upgraded.

**Comment by jpschewe** [comment]
This would also remove the extra dos box that comes up on Windows.

**Comment by jpschewe** [comment]
Some pointers https://www.eclipse.org/jetty/documentation/current/configuring-jsp.html and https://www.eclipse.org/jetty/documentation/current/embedded-examples.html#embedded-webapp-jsp

**Comment by jpschewe** [comment]
Example embedded jetty using jsps https://github.com/jetty-project/embedded-jetty-jsp

**Comment by jpschewe** [comment]
Some links on processing annotations. This is required to pickup which classes are servlets.

- https://www.eclipse.org/jetty/documentation/9.4.x/using-annotations.html
- https://www.eclipse.org/jetty/documentation/9.4.x/configuring-webapps.html#container-include-jar-pattern - probably need to set this property
- https://www.programcreek.com/java-api-examples/?api=org.eclipse.jetty.webapp.WebInfConfiguration
- https://www.eclipse.org/jetty/documentation/9.4.x/configuring-webapps.html

**Comment by jpschewe** [comment]
The end goal is to get rid of the system exec calls to start the webserver and stop the dos window from coming up on Windows. Trying tomcat-embed first as that may have fewer issues with the transition. I found a guide at https://devcenter.heroku.com/articles/create-a-java-web-application-using-embedded-tomcat. I've manually downloaded all of the dependencies (until issue #696 is done). I've added the libraries to the build and removed the external tomcat libraries from the build. That all builds. Now I need to try the launcher.
